eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -1,19 +1,27 @@
 parameters:
   Directory: 'not-specified'
-  IgnoreLinksFile: "$(Build.SourcesDirectory)/eng/ignore-links.txt"
-  
+  IgnoreLinksFile: '$(Build.SourcesDirectory)/eng/ignore-links.txt'
+  WorkingDirectory: '$(System.DefaultWorkingDirectory)'
+  ScriptDirectory: 'eng/common/scripts'
+  Recursive: $false
+  Branchreplaceregex: '$(System.PullRequest.SourceRepositoryURI)'
+  Branchreplacementname: '$(System.PullRequest.SourceCommitId)'
+  Checklinkguidance: $true
+  Suffixes: '*.md' # The suffix regex. Support multiple suffixes, separated by comma. e.g '*.md, *.html'
 steps:
-- task: PowerShell@2
-  displayName: Link verification check
-  inputs:
-    pwsh: true
-    workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
-    filePath: eng/common/scripts/Verify-Links.ps1 
-    arguments: >
-      -urls $(dir -r -i *.md) 
-      -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
-      -recursive: $false 
-      -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
-      -branchReplaceRegex "^($env:SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI.*/(?:blob|tree)/)master(/.*)$"
-      -branchReplacementName $env:SYSTEM_PULLREQUEST_SOURCECOMMITID
-      -devOpsLogging: $true
+  - task: PowerShell@2
+    displayName: Link verification check
+    inputs:
+      pwsh: true
+      workingDirectory: '${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}'
+      filePath: ${{ parameters.ScriptDirectory }}/Verify-Links.ps1
+      arguments: >
+        -suffixToScan ${{ parameters.Suffixes }}
+        -scanDir "${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}"
+        -rootUrl "file://${{ parameters.WorkingDirectory }}/${{ parameters.Directory }}"
+        -recursive: ${{ parameters.Recursive }} 
+        -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+        -branchReplaceRegex '^(${{ parameters.Branchreplaceregex }}.*/(?:blob|tree)/)master(/.*)$'
```

Comment:
Why is this only part of the regex?

Reply:
What about cases where someone would want to replace a feature branch name instead of master or something like that?